src/pivot-table/hooks/__tests__/use-selections-model.test.ts

```diff
@@ -59,24 +67,44 @@ describe("useSelectionsModel", () => {
 
   test("should select cell and not call begin selection when already active", async () => {
     selections.isActive = () => true;
-    const { result } = renderHook(() => useSelectionsModel(selections));
+    const { result } = renderHook(() => useSelectionsModel(selections, updatePageInfo));
 
     await act(async () => {
+      Object.entries(callbacks).forEach(([, cb]) => cb());
       await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 1)();
     });
 
     // eslint-disable-next-line @typescript-eslint/unbound-method
     await waitFor(() => expect(selections.begin).toHaveBeenCalledTimes(0));
+    await waitFor(() => expect(updatePageInfo).toHaveBeenCalledTimes(1));
     await waitFor(() => expect(result.current.isSelected(NxSelectionCellType.NX_CELL_TOP, 0, 1)).toBeTruthy());
   });
 
+  test("should select cell and mimic confirm action happening by calling related callback", async () => {
+    selections.isActive = () => true;
+    const { result } = renderHook(() => useSelectionsModel(selections, updatePageInfo));
+
+    await act(async () => {
+      callbacks.confirmed();
+      await result.current.select(NxSelectionCellType.NX_CELL_TOP, 0, 1)();
+    });
+
+    // trigger callback only if eventKey is "confirmed"
+    act(() => callbacks.confirmed());
+
+    // make sure it's been called 2 times
+    await waitFor(() => expect(updatePageInfo).toHaveBeenCalledTimes(2));
```

Comment:
Still, why call confirmed twice?

Reply:
because we want to simulate the confirm action happens after select call. 

we know that it happens useEffect once. 
but here, manually we do it one more time to say that after select call in line 89 confirm happens. 

then it's just a check to making sure that we call `updatePageInfo` 2 times  ( which is within the `clearSelectionAndResetPage` function that registered to confirmed event)

Reply:
No it is not called twice. The use effect calls `selection.on` that passes the callback to the listener, it does not run the callback. The callback runs when there is a confirm event, which we can simulate in the test now by `callbacks.confirmed()`.